### PR TITLE
CloudWatch Integration Tests

### PR DESCRIPTION
**Pull request by shurvitz**
**Description of changes:**
Added integration tests to the Cloudwatch connector.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java [hunk 1]
```diff
@@ -240,7 +240,7 @@ protected void cleanUp()
     public List<String> listTables(String databaseName)
             throws RuntimeException
     {
-        String query = String.format("show tables in %s.%s;", lambdaFunctionName, databaseName);
+        String query = String.format("show tables in `%s`.`%s`;", lambdaFunctionName, databaseName);
```
Comment (janmran) [inline athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java:243]: does backtick actually work? I though either not enclosed, or enclosed in double quotes.

File: athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java [hunk 2]
```diff
@@ -258,7 +258,7 @@ protected void cleanUp()
     public Map<String, String> describeTable(String databaseName, String tableName)
             throws RuntimeException
     {
-        String query = String.format("describe %s.%s.%s;", lambdaFunctionName, databaseName, tableName);
+        String query = String.format("describe `%s`.`%s`.`%s`;", lambdaFunctionName, databaseName, tableName);
```
Comment (janmran) [inline athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java:261]: same question here

File: athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java (hunk 1 above)
Comment (shurvitz) [reply athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java:243]: Only for DDL commands: `lambda:function`.`database`.`table`

File: athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java (hunk 2 above)
Comment (shurvitz) [reply athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/IntegrationTestBase.java:261]: Same as above. for DDL you can enclose in back-quote (needed for special characters: e.g. slash '/')

**Review by janmran — APPROVED** [review]

File: athena-cloudwatch/etc/test-config.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "spill_prefix" : "athena-spill",      /* The prefix within the S3 spill bucket (default: athena-spill) */
+    "disable_spill_encryption" : "false"  /* If set to true encryption for spilled data is disabled (default: false) */
+  },
+  "vpc_configuration" : {           /* This connector does not use a VPC configuration */
+    "vpc_id": "",                   /* Leave empty */
+    "security_group_id": "",        /* Leave empty */
+    "subnet_ids": [],               /* Leave empty */
+    "availability_zones": []        /* Leave empty */
+  },
+  "user_settings" : {                   /* User customizable settings */
```
[the remark this answers is not in the document]
Comment (shurvitz) [reply athena-cloudwatch/etc/test-config.json:15]: Anthony wanted to have generic templates that the Cx can use to capture certain variables used in the integration tests. The reason they are generic is that we don't know the specific configuration the Cx has for their account etc... The logic is that when we automate the the execution of the tests during code push, we will overwrite the templates with config files specific to our configuration.

**Pull request merged**